File: 1/part1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

using namespace std;

ifstream infile("input.txt");

string line;

int main() {
  int max_calories = 0;
  int current = 0;
  while (getline(infile, line))
  {
    line.erase(remove(line.begin(), line.end(), '\r'), line.cend());

    if (line.empty())
    {
      max_calories = max(max_calories, current);
      current = 0;
      continue;
    }
    current += stoi(line);
  }
  std::cout << "Max calories: " << max_calories << std::endl;
}

File: 1/part1.rs
// Advent of code 2022 - day 1

use std::fs::*;

fn main() {
  let input = read_to_string("input.txt").unwrap();
  let lines = input.lines();

  let mut max_calories = 0;
  let mut current = 0;
  for line in lines {
    if line.is_empty() {
      max_calories = max_calories.max(current);
      current = 0;
      continue;
    }

    let calories = line.parse::<i32>().unwrap();
    current += calories;
  }
  
  // Print the maximum calories
  println!("Max calories: {}", max_calories);
}

